### DISCUSSION TITLE
Cannot use CreateObjectURL because it violates Content Security Policy directive

### DISCUSSION THREAD

**Discussion by goldenpath-ai** [directus/directus#15448] [Archive]
Hello, I have a custom directus module in which a user uploads a pdf document. I need to access the document in pdf.js, so I need to create an object URL. However, I cannot seem to get it to work. How do I set the CSP directive to something that allows for this or is there a workaround?

`Refused to connect to 'blob:http://localhost:8055/905f08e2-4801-4797-bccb-826980f734e1' because it violates the following Content Security Policy directive: "connect-src 'self' https://*".`

**Comment by rijkvanzanten** [comment]
You can update the full `helmet` configuration through the `CONTENT_SECURITY_POLICY_*` environment variables.
